src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/pdf/util/PdfBoxServiceUtil.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.service.pdf.util;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static uk.gov.hmcts.ethos.replacement.docmosis.service.pdf.PdfBoxServiceConstants.CHARACTER_DOT;
+import static uk.gov.hmcts.ethos.replacement.docmosis.service.pdf.PdfBoxServiceConstants.CURRENCY_DECIMAL_ZERO_WITH_DOT;
+import static uk.gov.hmcts.ethos.replacement.docmosis.service.pdf.PdfBoxServiceConstants.ONE;
+import static uk.gov.hmcts.ethos.replacement.docmosis.service.pdf.PdfBoxServiceConstants.STRING_ZERO;
+import static uk.gov.hmcts.ethos.replacement.docmosis.service.pdf.PdfBoxServiceConstants.TWO;
+import static uk.gov.hmcts.ethos.replacement.docmosis.service.pdf.PdfBoxServiceConstants.ZERO;
+import static uk.gov.hmcts.ethos.replacement.docmosis.service.pdf.et3.ET3FormConstants.STRING_EMPTY;
+
+@Slf4j
+public final class PdfBoxServiceUtil {
+
+    private PdfBoxServiceUtil() {
+        // Utility classes should not have a public or default constructor.
+    }
+
+    /**
+     * Logs exception according to given user inputs below. It enables us have a better log on azure cloud.
+     * @param firstWord first wordings of the log.
+     * @param caseReferenceNumber ethos case reference data
+     * @param message exception message
+     * @param className class name where the exception occurs
+     * @param methodName method name where the exception occurs
+     */
+    public static void logException(String firstWord,
+                                    String caseReferenceNumber,
+                                    String message,
+                                    String className,
+                                    String methodName) {
+        log.error("*************EXCEPTION OCCURED*************"
+                + "\nERROR DESCRIPTION: " + (isNotBlank(firstWord) ? firstWord : STRING_EMPTY)
+                + "\nCASE REFERENCE: " + (isNotBlank(caseReferenceNumber) ? caseReferenceNumber : STRING_EMPTY)
+                + "\nERROR MESSAGE: " + (isNotBlank(message) ? message : STRING_EMPTY)
+                + "\nCLASS NAME: " + (isNotBlank(className) ? className : STRING_EMPTY)
+                + "\nMETHOD NAME: " + (isNotBlank(methodName) ? methodName : STRING_EMPTY)
+                + "\n*****************END OF EXCEPTION MESSAGE***********************");
+    }
```

Comment:
this seems very similar to what PdfBoxServiceException is doing so can this be removed/replaced by that version?

Reply:
Harpreet, when I want to throw an exception, I use the class that I generated as PdfBoxServiceException but whenever I just want to log I use logException. For this case I removed the repeating code in the exception class and called loggingUtil.logexception method and renamed PdfBoxServiceException as GenericServiceException and moved to package uk.gov.hmcts.ethos.replacement.docmosis.exceptions.